app/uk/gov/hmrc/agentsexternalstubsfrontend/controllers/IdentityVerificationController.scala

```diff
@@ -28,60 +32,124 @@ class IdentityVerificationController @Inject()(
+                options(journeyIdValue),
+                routes.IdentityVerificationController
+                  .uplift(journeyIdValue, confidenceLevel, completionURL, failureURL, origin)
+              ))
+        }
+      }
     }
 
-  def uplift(completionURL: ContinueUrl, failureURL: ContinueUrl): Action[AnyContent] =
+  def uplift(
+    journeyId: String,
+    confidenceLevel: Int,
+    completionURL: ContinueUrl,
+    failureURL: ContinueUrl,
+    origin: Option[String]): Action[AnyContent] =
     Action.async { implicit request =>
       authorised()
         .retrieve(Retrievals.credentialsWithPlanetId) { credentials =>
           UpliftRequestForm
             .bindFromRequest()
             .fold(
               formWithErrors =>
-                Future.successful(Ok(html
-                  .iv_uplift(formWithErrors, routes.IdentityVerificationController.uplift(completionURL, failureURL)))),
+                Future.successful(
+                  Ok(
+                    html
+                      .iv_uplift(
+                        formWithErrors,
+                        options(journeyId),
+                        routes.IdentityVerificationController
+                          .uplift(journeyId, confidenceLevel, completionURL, failureURL, origin)))),
               upliftRequest => {
-                if (upliftRequest.willSucceed) {
+                val journeyIdMatches = upliftRequest.option.contains(journeyId)
+                val isSuccessful = upliftRequest.option.contains("success")
+                if (journeyIdMatches && isSuccessful) {
                   for {
                     currentUser <- agentsExternalStubsConnector.getUser(credentials.providerId)
-                    modifiedUser = currentUser.copy(confidenceLevel = Some(upliftRequest.confidenceLevel))
+                    modifiedUser = currentUser.copy(confidenceLevel = Some(200))
                     _ <- agentsExternalStubsConnector.updateUser(modifiedUser)
-                  } yield Redirect(completionURL.url)
+                  } yield redirectWithJourneyId(completionURL.url, journeyId, upliftRequest.option)
                 } else {
-                  Future.successful(Redirect(failureURL.url))
+                  Future.successful(redirectWithJourneyId(failureURL.url, journeyId, upliftRequest.option)
+                    .withHeaders("X-Result-Location" -> s"/mdtp/journey/journeyId/${upliftRequest.option}"))
                 }
               }
             )
         }
     }
 
+  private def redirectWithJourneyId(targetUrl: String, journeyId: String, option: String): Result = {
+    val cleanUrl = URLDecoder.decode(targetUrl, "UTF-8")
+    val tokenAppendChar = if (cleanUrl.contains("?")) "&" else "?"
+    Redirect(s"$cleanUrl${tokenAppendChar}journeyId=$journeyId")
+      .withHeaders("X-Result-Location" -> s"/mdtp/journey/journeyId/$option")
+  }
+
+  def getIvResult(journeyIdAndReason: String): Action[AnyContent] = Action.async { implicit request =>
```

Comment:
do we need to have a getIveResultInternal for staging?

Reply:
Ah fair point